### PR TITLE
Issue #34 Add logic to handle metadata of tarball

### LINKED ISSUE #98: Check driver version during pre-flight
Currently the driver implements a version but is not yet enforced by the pre-flights. When a change to the bundling/metadata happens, the driver also gets updated.

### PR DESCRIPTION
**Pull request by praveenkumar**
- Update the crc-libvirt-driver version
- Remove unused domain stuff part of libvirt/driver part
- Fix #98

### CODE REVIEW

File: pkg/crc/machine/libvirt/driver.go [hunk 1]
```diff
@@ -23,9 +22,11 @@ type libvirtDriver struct {
 	DiskPath  string
 	CacheMode string
 	IOMode    string
+	DiskPathUrl string
+	SSHKeyPath  string
 }
 
-func CreateHost(machineConfig config.MachineConfig) *libvirtDriver {
+func CreateHost(machineConfig config.MachineConfig, ) *libvirtDriver {
```
Comment (gbraad) [inline pkg/crc/machine/libvirt/driver.go]: why the `, ` ?

File: pkg/crc/machine/bundle/extract.go [hunk 2]
```diff
@@ -1,29 +1,36 @@
...
+func Extract(sourcepath string, destpath string) (string, error) {
 	file, err := os.Open(sourcepath)
 
 	if err != nil {
-		return err
+		return "", err
+	}
+
+	extractedPath := strings.Split(filepath.Base(file.Name()), ".tar")[0]
+	extractedPath = filepath.Join(destpath, extractedPath)
```
Comment (gbraad) [inline pkg/crc/machine/bundle/extract.go]: so what if someone changes the name of the bundle?

Comment (gbraad) [reply pkg/crc/machine/bundle/extract.go]: This info should come from the file type that gets extracted. it contains the actual path and this needs to be split... and not the archive file name.

File: pkg/crc/machine/libvirt/driver.go [hunk 3]
```diff
@@ -36,9 +37,10 @@ func CreateHost(machineConfig config.MachineConfig) *libvirtDriver {
 		Memory:     machineConfig.Memory,
 		CPU:        machineConfig.CPUs,
 		Network:    DefaultNetwork,
-		// DiskPath should come from the bundle's metadata (unflattened)
-		DiskPath:  filepath.Join(constants.MachineBaseDir, "machines", machineConfig.Name, "crc_libvirt_0.16.1", "crc"),
 		CacheMode: DefaultCacheMode,
 		IOMode:    DefaultIOMode,
+		DiskPath:  filepath.Join(constants.MachineBaseDir, "machines", machineConfig.Name, constants.DefaultName),
```
Comment (gbraad) [inline pkg/crc/machine/libvirt/driver.go]: why both?

File: pkg/crc/preflight/preflight_checks_linux.go [hunk 4]
```diff
@@ -37,6 +37,7 @@ var (
 	crcDnsmasqConfig        = `server=/tt.testing/192.168.126.1
 address=/apps.tt.testing/192.168.126.11
 `
+	libvirtDriverDownloadURL = fmt.Sprintf("https://github.com/code-ready/machine-driver-libvirt/releases/download/%s/crc-driver-libvirt", libvirtDriverVersion)
```
Comment (gbraad) [inline pkg/crc/preflight/preflight_checks_linux.go]: `var` ... I guess so...

File: vendor/modules.txt [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# github.com/code-ready/machine v0.7.1-0.20190424064536-bab5ede5500e
+# github.com/code-ready/machine v0.7.1-0.20190501073902-87391ca2ceed
```
Comment (gbraad) [inline vendor/modules.txt]: ??

File: vendor/modules.txt [hunk 6]
```diff
@@ -59,6 +59,8 @@ github.com/spf13/jwalterweatherman
 github.com/spf13/pflag
 # github.com/spf13/viper v1.3.2
 github.com/spf13/viper
+# github.com/xi2/xz v0.0.0-20171230120015-48954b6210f8
```
Comment (gbraad) [inline vendor/modules.txt]: and why also this ??

File: vendor/modules.txt (hunk 5 above)
Comment (praveenkumar) [reply vendor/modules.txt]: These are module txt, you can ignore.

**Comment by anjannath** [comment]
_Locally tested the PR, works for me._

**Pull request merged**